internal/encoding/parquet/parquet.go

```diff
@@ -111,13 +112,44 @@ func (i *iterator) Schema() typeof.Schema {
 	for _, c := range schema.Columns() {
 		t := c.Type()
 
-		if t, supported := typeof.FromParquet(t); supported {
-			result[c.Name()] = t
+		if t == nil {
```

Comment:
Move this `t == nil` and `supported` check into `getSchemaLogicalType` function instead.